students/arkadiusz_kasprzyk/lesson_01_basics/car_route.py

```diff
@@ -0,0 +1,29 @@
+'''
+title: car_route
+author: arkadiusz.kasprzyk@tieto.com
+date: 2018-03-05
+description:
+    A car can cover distance of N kilometers per day.
+    How many days will it take to cover a route of length M kilometers?
+    The program gets two numbers: N and M.
+'''
+
+import math as m
+tol = 1e9  ## tolerance
+
+print('''
+    A car can cover distance of N kilometers per day.
+    How many days will it take to cover a route of length M kilometers?
+    '''
+)
+
+N = float(input("Number of km per day: "))
+M = float(input("Number of km to cover: "))
+
+f = m.floor(M/N)
+r = f + int(abs(M-f) > tol)
```

Comment:
It doesn't work because `-` sign disappeared: there should be `tol = 1e-9` not `1e9`. Just typo. 
I agree that `ceil` is better. Just fixation on using `floor`.
By the way, what's the general tactic to avoid usig tolerance?

Reply:
I'm no expert in numerical methods, but the general rule is that whenever floating points approach zero or infinity it gets unsafe: they may change sign, are unreliable when compared or cause big errors when multiplied. The idea is to keep the numbers close to the middle of the range. I just try to use libraries and do unit tests with some edge cases, because I rarely deal with that and only dig down deeper, if something goes wrong.